### PR TITLE
Fix tests after bundler change - Stacked 1/3

### PR DESCRIPTION
**Pull request by schneems**
The Ruby buildpack uses the Ruby buildpack on CI. When the bundler versions changed in https://github.com/heroku/heroku-buildpack-ruby/commit/56ff138eea331620d3579370693dd39140a642cd it also changed the behavior of these tests (which use the local bundler version). That means that after the buildpack was deployed, it retroactively meant that tests on main were failing.

The issue with these tests is that they're using a feature of bundler that's been removed, specifically this change https://devcenter.heroku.com/changelog-items/2809. In these tests the Ruby version is specified in the Gemfile but not the Gemfile.lock:

- Gemfile https://github.com/sharpstone/mri_193/blob/master/Gemfile
- Gemfile.lock https://github.com/sharpstone/mri_193/blob/master/Gemfile.lock 

With the newer version of bundler, these tests now fail. To fix this tests will need to be updated or removed (based on their current relevance).

### CODE REVIEW

File: spec/helpers/ruby_version_spec.rb [hunk 1]
```diff
@@ -60,20 +60,6 @@
...
-        version_number = "1.9.3"
-        version        = "ruby-#{version_number}"
-        expect(ruby_version.version_without_patchlevel).to eq(version)
-        expect(ruby_version.engine_version).to eq(version_number)
-        expect(ruby_version.to_gemfile).to eq("ruby '#{version_number}'")
-        expect(ruby_version.engine).to eq(:ruby)
-      end
-    end
-  end
-
```
Comment (schneems) [inline spec/helpers/ruby_version_spec.rb:76]: 1.9.3 is no longer supported on the platform, we can remove this test

File: spec/helpers/ruby_version_spec.rb [hunk 2]
```diff
@@ -103,11 +89,39 @@
...
+
+          RUBY VERSION
+            ruby 3.2.3p157
+
+          BUNDLED WITH
+            2.4.19
+        EOF
+
         ruby_version   = LanguagePack::RubyVersion.new(@bundler.install.ruby_version, is_new: true)
-        version_number = "2.0.0"
```
Comment (schneems) [inline spec/helpers/ruby_version_spec.rb:110]: Update test to use a modern Ruby version and move away from the less flexible github based fixture. Write the Gemfile and Gemfile.lock instead so the developer can see what's being used in the test at a glance.

File: spec/helpers/ruby_version_spec.rb [hunk 3]
```diff
@@ -117,32 +131,42 @@
...
+        ruby_version   = LanguagePack::RubyVersion.new(
+          @bundler.install.ruby_version,
+          is_new: true
+        )
+        version_number = "2.6.8"
+        engine_version = "9.3.6.0"
+        engine = :jruby
+        expect(ruby_version.version_without_patchlevel).to eq("ruby-#{version_number}-#{engine}-#{engine_version}")
         expect(ruby_version.engine_version).to eq(engine_version)
-        expect(ruby_version.to_gemfile).to eq(to_gemfile)
```
Comment (schneems) [inline spec/helpers/ruby_version_spec.rb:132]: Same as above, update to not rely on a github template. Update to use a more recent Ruby version

File: spec/helpers/ruby_version_spec.rb [hunk 4]
```diff
@@ -117,32 +131,42 @@
...
-  it "surfaces error message from bundler"  do
-    bundle_error_msg = "Zomg der was a problem in da gemfile"
-    error_klass      = LanguagePack::Helpers::BundlerWrapper::GemfileParseError
-    Hatchet::App.new("bad_gemfile_on_platform").in_directory_fork do |dir|
-      Bundler.with_unbundled_env do
-        @bundler       = LanguagePack::Helpers::BundlerWrapper.new().install
-        expect { LanguagePack::RubyVersion.new(@bundler.ruby_version) }.to raise_error(error_klass, /#{Regexp.escape(bundle_error_msg)}/)
-      end
-    end
-  end
```
Comment (schneems) [inline spec/helpers/ruby_version_spec.rb:147]: This test is supposed to exercise this code https://github.com/heroku/heroku-buildpack-ruby/blob/7866d1efbdc0e4d5f66caff2cdd7f83f76d4f8e0/lib/language_pack/helpers/bundler_wrapper.rb#L176

However bundler no longer errors on `bundle platform --ruby` depending on what version of bundler is running.

In the future we will move to get rid of the concept of "blessed" bundler versions and to BYOBv (bring your own bundler version) which means all that code will eventually go away. I don't want to remove the check and re-raise logic right now though. If we wanted to test this functionality we would need to move away from these local "unit-ish" tests and to a full blown hatchet test that tries to deploy an app.

**Review by joshwlewis — APPROVED** [review]
Nice. Thanks for extracting. This totally makes sense in isolation.

**Comment by schneems** [comment]
It was merged in with https://github.com/heroku/heroku-buildpack-ruby/pull/1440 which was also approved. I clicked the "enable auto merge" but didn't see that my own comments were considered blocking so it wasn't merged in. That PR had the exact same commits anyway, so it's the exact same code. Leaving a note as to why I'm closing this after approval.

**Pull request closed without merging**